### DISCUSSION TITLE
How should you validate the payload?

### DISCUSSION THREAD

**Discussion by jansgescheit** [spatie/laravel-webhook-client#188] [Q&A]
Does anyone have an extended example of this lib? 
I wonder where I can validate the payload of a hook. If I were to write a webhook handling myself, I would have placed this classically in the controller as a parameter.

**Comment by freekmurze** [comment]
Here's how to validate the signature of an incoming webhook: https://github.com/spatie/laravel-webhook-client#creating-your-own-signature-validator

Verifying the content itself could be done in the job that handles the webhook (that it out of scope for the package)
